samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/assessment/AssessmentEntityProducer.java

```diff
@@ -425,68 +446,51 @@ public void updateEntityReferences(String toContext, Map<String, String> transve
-						if(itemTextList != null){
-							for(int k = 0; k < itemTextList.size(); k++){
-								ItemText itemText = (ItemText) itemTextList.get(k);
-								String text = itemText.getText();
-								if(text != null){
-									// Transfer all of the attachments to the new site
-									text = service.copyContentHostingAttachments(text, toContext);
-
-									text = org.sakaiproject.util.cover.LinkMigrationHelper.migrateAllLinks(entrySet, text);
-									if(!text.equals(itemText.getText())){
-										//need to save since a ref has been updated:
-										needToUpdate = true;
-										itemText.setText(text);
-									}
-								}
+						boolean instructionChanged = migrateText(service, toContext, item, itemHash, hasCaches, hasDuplicates, false,
+								"inst", itemContentCache, entrySet, ItemData::getInstruction, ItemData::setInstruction);
 
-								List answerSetList = itemText.getAnswerArray();
-								if (answerSetList != null) {
-									for (int l = 0; l < answerSetList.size(); l++) {
-										Answer answer = (Answer) answerSetList.get(l);
-										String answerText = answer.getText();
+						boolean descriptionChanged = migrateText(service, toContext, item, itemHash, hasCaches, hasDuplicates, false,
+								"desc", itemContentCache, entrySet, ItemData::getDescription, ItemData::setDescription);
 
-										if (answerText != null) {
-											// Transfer all of the attachments embedded in the answer text
-											answerText = service.copyContentHostingAttachments(answerText, toContext);
+						boolean itemTextsChanged = false;
+						List<ItemTextIfc> itemTexts = item.getItemTextArray();
+						if (itemTexts != null) {
+							for (ItemTextIfc itemText : itemTexts) {
+								boolean itemTextChanged = migrateText(service, toContext, itemText, itemHash, hasCaches, hasDuplicates, true,
+										"it-" + itemText.getSequence(), itemContentCache, entrySet, ItemTextIfc::getText, ItemTextIfc::setText);
 
-											// Now rewrite the answerText with links to the new site
-											answerText = org.sakaiproject.util.cover.LinkMigrationHelper.migrateAllLinks(entrySet, answerText);
+								boolean answersChanged = false;
+								List<AnswerIfc> answers =  itemText.getAnswerArray();
+								if (answers != null) {
+									for (AnswerIfc answer : answers) {
+										boolean answerChanged = migrateText(service, toContext, answer, itemHash, hasCaches, hasDuplicates, true,
+												"at-" + itemText.getSequence() + "-"+ answer.getSequence() , itemContentCache, entrySet, AnswerIfc::getText, AnswerIfc::setText);
 
-											if (!answerText.equals(answer.getText())) {
-												needToUpdate = true;
-												answer.setText(answerText);
-											}
-										}
+										answersChanged = answersChanged || answerChanged;
 									}
 								}
+
+								itemTextsChanged = itemTextsChanged || itemTextChanged || answersChanged;							
 							}
 						}
+
+						needToUpdate = needToUpdate
```

Comment:
Just a question about this, is needToUpdate inside the "items" loop? I'm not sure I understand it and I'm concerned it carries its value to the next one.

Reply:
You are right, I fixed it